memory.tex

```diff
@@ -1773,21 +1721,103 @@ \subsection{C64-style Memory Mapping}
+access RAM at 0.0000 and 0.0001, and will not access the C64-style memory
+mapping registers. The C64-style banking registers can only be accessed with
+the \$0000 -- \$1FFF region unmapped.
+
+It is typical for a MEGA65 program to leave I/O registers banked into \$D000,
+and use the MAP register when it needs to access the underlying RAM. Such a
+program does not need to manipulate the C64-style banking register.
+
+\subsection{The \$D030 VIC-III Banking Register}
+
+The Commodore 65 provides a banking mechanism controlled by the VIC chip to
+make ROM data accessible with 16-bit addresses. This banking is controlled by the
+I/O register \$D030.\footnote{Accessing the VIC-III banking register at \$D030 requires
+that the computer be using the VIC-III/IV ``I/O personality,'' described
+later. This is the default in MEGA65 mode.}
+
+Bits 3, 4, 5, and 7 map regions of ROM data into the 16-bit address space according to
+this table. (Recall that bits are numbered right to left from 0, so bit 7 is the
+highest bit.)
 
-\subsection{Cartridge ROM}
+\begin{center}
+\begin{longtable}{|L{1.5cm}|L{1.5cm}|L{1.5cm}|L{1.5cm}|}
+\hline
+{\bf \$D030 Bit} & {\bf Signal Name} & {\bf 20-bit Address} & {\bf 16-bit Address} \\
+\hline
+\endfirsthead
+\multicolumn{4}{l@{}}{\ldots continued}\\
+\hline
+{\bf \$D030 Bit} & {\bf Signal Name} & {\bf 20-bit Address} & {\bf 16-bit Address} \\
+\endhead
+\multicolumn{4}{l@{}}{continued \ldots}\\
+\endfoot
+\hline
+\endlastfoot
+\hline
+3 & ROM8 & \$28000 -- \$29FFF & \$8000 -- \$9FFF \\
+\hline
+4 & ROMA & \$2A000 -- \$2BFFF & \$A000 -- \$BFFF \\
+\hline
+5 & ROMC & \$2C000 -- \$2CFFF & \$C000 -- \$CFFF \\
+\hline
+7 & ROME & \$2E000 -- \$2FFFF & \$E000 -- \$FFFF \\
+\end{longtable}
+\end{center}
+
+For example, if \$D030 is set to \$64 = ``0 1 1 0 0 1 0 0,'' then ROMC is
+enabled, and the 16-bit addresses \$C000 -- \$DFFF access 2.C000 -- 2.DFFF.
+
+As with C64-style banking, \$D030 banking only applies to 8KB address blocks not
+selected for offsetting by the MAP register. For unmapped regions, if either
+C64-style ROM banking or \$D030 ROM banking is enabled for the region,
+then those addresses refer to ROM in bank 2. Each of these ROM banking
+mechanisms use ROM in bank 2, so there is no conflict if they are both
+enabled.
 
-When a cartridge is inserted in the expansion port, the memory system overrides
-regions of the 16-bit address space to access lines to the cartridge, typically
-connected to ROM chips in the cartridge. The cartridge controls two additional
-lines that allow it to request different memory configurations using the 0.0000
-and 0.0001 CPU registers. These configurations are identical to that of a
-Commodore 64. See a C64 reference book for more information.
+Notice that \$D030 does not bank in any of the MEGA65 kernel ROM from bank 3.
+The kernel itself uses the MAP register to access its own code, and only uses
+\$D030 banking to access the PETSCII character set at 2.C000 with ROMC
```

Comment:
$2.C000 contains the interface ROM, not the PETSCII character set.

Reply:
Fixed.